tests/DependencyInjection/Compiler/ModelManagerCompilerPassTest.php

```diff
@@ -16,48 +16,157 @@
 use PHPUnit\Framework\TestCase;
 use Prophecy\Argument;
 use Sonata\AdminBundle\DependencyInjection\Compiler\ModelManagerCompilerPass;
-use Sonata\AdminBundle\Maker\AdminMaker;
+use Sonata\AdminBundle\Model\ModelManagerInterface;
+use Sonata\AdminBundle\Tests\App\Model\ModelManager;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
 use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Exception\LogicException;
 
 /**
  * @author Gaurav Singh Faujdar <faujdar@gmail.com>
  */
-class ModelManagerCompilerPassTest extends TestCase
+final class ModelManagerCompilerPassTest extends TestCase
 {
+    public function testProcess(): void
+    {
+        $adminMaker = $this->prophesize(Definition::class);
+        $adminMaker->replaceArgument(Argument::type('integer'), Argument::any())->shouldNotBeCalled();
+        $containerBuilderMock = $this->prophesize(ContainerBuilder::class);
+
+        $containerBuilderMock->getServiceIds()
+            ->willReturn([]);
+
+        $containerBuilderMock->findTaggedServiceIds(Argument::exact(ModelManagerCompilerPass::MANAGER_TAG))
+            ->willReturn([]);
+
+        $containerBuilderMock->getParameter(Argument::exact('kernel.bundles'))
+            ->willReturn(['MakerBundle' => 'MakerBundle']);
+
+        $containerBuilderMock->getDefinition(Argument::exact('sonata.admin.maker'))
+            ->willReturn($adminMaker->reveal());
+        $containerBuilderMock->getParameter(Argument::containingString('kernel.project_dir'))
+            ->willReturn(null);
+
+        $compilerPass = new ModelManagerCompilerPass();
+        $compilerPass->process($containerBuilderMock->reveal());
+    }
+
     /**
-     * @var AdminMaker
+     * NEXT_MAJOR: Remove this test.
+     *
+     * @group legacy
+     *
+     * @expectedDeprecation Using the "sonata.admin.manager" tag in the service "sonata.admin.manager.test" is required since sonata-project/admin-bundle 3.x.
```

Comment:
```suggestion
     * @expectedDeprecation Not setting the "sonata.admin.manager" tag on the "sonata.admin.manager.test" service is deprecated since sonata-project/admin-bundle 3.x.
```